models/classes/class.QtiTestConverter.php

```diff
@@ -315,18 +321,22 @@ private function arrayToComponent(array $testArray, QtiComponent $parent = null,
      * Get the value according to it's type and class.
      *
      * @param mixed $value
-     * @param object $class
-     * @return \qtism\common\datatypes\QtiDuration
+     * @param object|null $class
+     * @return QtiDuration|QtiComponentCollection|mixed|null
      */
     private function componentValue($value, $class)
     {
-        if (! is_null($class)) {
-            if (is_array($value)) {
-                return $this->createComponentCollection(new ReflectionClass($class->name), $value);
-            } elseif ($class->name === 'qtism\common\datatypes\QtiDuration') {
-                return new qtism\common\datatypes\QtiDuration('PT' . $value . 'S');
-            }
+        if ($class === null) {
+            return $value;
         }
+
+        if (is_array($value)) {
+            return $this->createComponentCollection(new ReflectionClass($class->name), $value);
+        }
+        if ($class->name === QtiDuration::class) {
+            return new QtiDuration('PT' . $value . 'S');
```

Comment:
a bit weird to see this special case here

Reply:
This case was here, I just fixed a bit of readability 🙂